TUnit.Core.SourceGenerator/Generators/TestMetadataGenerator.cs

```diff
@@ -36,6 +36,14 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             .Where(static m => m is not null)
             .Combine(enabledProvider);
 
+        // Custom test attributes that inherit from BaseTestAttribute
+        var customTestMethodsProvider = context.SyntaxProvider
```

Comment:
I'm cautious about this part because these syntax providers that trigger on code can create a very laggy experience and fire a lot - That's why they recommend just going with the `ForAttributeWithMetadataName`.

I think keeping the `[Test]` attribute isn't a bad thing anyway - It's then always a marker for something executable.

The new property one can remain and just be the event receiver?

What's do you think?

Reply:
Yeah, I had the same thought, was going to suggest that as an option.

Another option (not sure if it's feasible) is to try to discover all of the inheriting attributes upfront, and then  listening for those with `ForAttributeWithMetadataName`.

Reply:
I think you'd need access to the compilation syntax then to do a type lookup, which is just as expensive, and compilation syntax fires on literally every single keystroke on anything. I don't think there's a nice performant way without using predefined well known attributes 😢

Reply:
Ah yeah, I guess the other option is to move some of the source generation to `TUnit.FsCheck`.

But right now I think it makes sense to try it with both attributes.